crates/bevy_winit/src/lib.rs

```diff
@@ -772,13 +788,28 @@ fn handle_winit_event(
         Event::UserEvent(RequestRedraw) => {
             runner_state.redraw_requested = true;
         }
+        Event::LoopExiting => {
+            let exit = exit_happened.clone().unwrap_or_else(|| {
```

Comment:
I think this is unreachable. We only receive this event if we call `event_loop.exit()`, which is guaranteed to have set `*exit_happened = Some(app_exit)`.

Reply:
The event drop code specifically let's through the `LoopExiting` event.
https://github.com/bevyengine/bevy/blob/851b21af39c701496e74bc88e5e63f46185e7b0a/crates/bevy_winit/src/lib.rs#L364-L367

I didn't put much thought into my solution as I know this code will get replaced soon<sup>TM</sup>. I'd modify your solution to drop events after we start exiting the event loop and merge it instead of this one. That's not my call to make though.

Reply:
I'm just saying we'll only receive `LoopExiting` if we've already called `event_loop.exit()` which in the control flow here guarantees `exit_happened` is `Some`. Agree that this code will likely change a lot post non-send changes.